tests/modules/targets-linux/generic-linux-gpu/nvidia-enabled.nix

```diff
@@ -0,0 +1,54 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  mockNvidiaDriver = pkgs.runCommand "mock-nvidia-driver" { } ''
+    mkdir -p $out/lib
+    echo "MOCK_NVIDIA_DRIVER_FOR_TESTING" > $out/lib/nvidia-test-marker
+  '';
+
+  # Override the package set to provide our mock driver. This also tests that a
+  # custom package set can be used with this module. The mock driver needs to
+  # support .override because the module calls it.
+  customPkgs = pkgs // {
+    linuxPackages = pkgs.linuxPackages // {
+      nvidiaPackages = pkgs.linuxPackages.nvidiaPackages // {
+        mkDriver = args: lib.makeOverridable (_: mockNvidiaDriver) { };
+      };
+    };
+  };
```

Comment:
Suggestion: having a `finalPackage` read-only option might make it easier to override with a mock driver package?

Reply:
I think we could do something like `nvidiaPackage = lib.mkForce mkStub ...` in the tests, which might be _slightly_ easier.

It wouldn't really provide any value apart from that.